internal/credit/postgres_connector/balance.go

```diff
@@ -190,7 +194,7 @@ func (a *PostgresConnector) getBalance(
 			if expiresAt.Before(period.From) {
 				continue
 			}
-			if grantBalance.EffectiveAt.After(period.To) {
+			if grantBalance.EffectiveAt.After(period.To.Truncate(time.Minute)) {
```

Comment:
Let's add this to the docs that right now we have a 1 minute resolution.

cc @hekike please be aware of this limitation